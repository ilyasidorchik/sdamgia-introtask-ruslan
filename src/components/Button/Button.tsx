import React, { ReactNode, FC } from 'react'
import { cn } from '@bem-react/classname'

import './Button.scss'

interface IButtonProps {
  className?: string
  disabled?: boolean
  children: ReactNode
}

const Button: FC<IButtonProps> = ({
  className,
  disabled = false,
  children,
}) => {
  const cnButton = cn('Button')

  return (
    <button className={cnButton({ disabled }, [className])} disabled={disabled}>
      {children}
    </button>
  )
}

export default Button
