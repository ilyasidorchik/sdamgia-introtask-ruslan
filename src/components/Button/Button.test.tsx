import React from 'react'
import { render } from '@testing-library/react'

import Button from './Button'

describe('Button', () => {
  describe('common', () => {
    const { container } = render(<Button>Найти</Button>)
    const component = container.querySelector('.Button')

    it('renders correctly', () => {
      expect(component).toHaveTextContent('Найти')
    })

    it('matches snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with extra className and disabled', () => {
    const { container } = render(
      <Button className={'Profile-Button'} disabled={true}>
        Сохранить
      </Button>
    )
    const component = container.querySelector('.Button.Button_disabled')

    it('renders correctly', () => {
      expect(component).toHaveTextContent('Сохранить')
    })

    it('matches snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
