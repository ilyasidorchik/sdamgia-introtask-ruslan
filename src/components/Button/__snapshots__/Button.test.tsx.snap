// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button common matches snapshot 1`] = `
<button
  class="Button"
>
  Найти
</button>
`;

exports[`Button with extra className and disabled matches snapshot 1`] = `
<button
  class="Button Button_disabled Profile-Button"
  disabled=""
>
  Сохранить
</button>
`;
